esp/services/ws_fs/ws_fsService.cpp

```diff
@@ -2278,6 +2278,55 @@ bool CFileSprayEx::onReplicate(IEspContext &context, IEspReplicate &req, IEspRep
     return true;
 }
 
+const char* CFileSprayEx::getDropZoneDirByIP(const char* ip, StringBuffer& dir)
+{
+    if (!ip || !*ip)
+        return NULL;
+
+    Owned<IEnvironmentFactory> factory = getEnvironmentFactory();
+    Owned<IConstEnvironment> env = factory->openEnvironment();
+    if (!env)
+        return NULL;
+
+    Owned<IPropertyTree> root = &env->getPTree();
+    VStringBuffer xPath("Hardware/Computer[@netAddress='%s']/@name", ip);
+    const char* computer = root->queryProp(xPath.str());
+    if (!computer || !*computer)
+    {
+        StringBuffer localHostIP;
+        queryHostIP().getIpText(localHostIP);
+        if (!strieq(".", ip) && !strieq("localhost", ip) && !strieq("127.0.0.1", ip) && !strieq(ip, localHostIP.str()))
+            return NULL;
+
+        StringArray possibleIPs;
+        if (!strieq(".", ip))
+            possibleIPs.append(".");
+        if (!strieq("localhost", ip))
+            possibleIPs.append("localhost");
+        if (!strieq("127.0.0.1", ip))
+            possibleIPs.append("127.0.0.1");
+        if (!strieq(localHostIP.str(), ip))
+            possibleIPs.append(localHostIP.str());
```

Comment:
The conditions here are to avoid re-looking up one of these candidates again, since it will have done so for one on line 2293 ?
I wouldn't bother, the code will be a lot clearer if you just look up all the equivalents, until you find a match.

Again, this should probably be contained in environment.* , something like getMachineForLocalHost()